File: express/src/controller/UserController.ts
import {NextFunction, Request, Response} from "express";
import { User } from '../entity/User';
import { iUserService } from '../service/UserService';

export class UserController {
    private service: iUserService;
    constructor(service: iUserService) {
        this.service = service;
    }

    all = async (req: Request<{ page: number}, any, any, { size: number, filt: string, ordr: string, dir: boolean}>, res: Response, next: NextFunction) => {
        const { page } = req.params;
        const { size, filt, ordr, dir } = req.query;
        
        try {
            const users = await this.service.findByPageAndSizeAndFilterAndOrder(page, size, filt, ordr, dir);
            return res.json({users: users[0], count: users[1]});
        } catch(err) {
            console.error('in all user:\n', err);
            next(err);
        }
    }

    one = async (req: Request<{id: number}>, res: Response, next: NextFunction) => {
        const { id } = req.params;

        try {
            const user = await this.service.findOne(id);
            return res.json({user});
        } catch(err) {
            console.error('in one user:\n', err);
            next(err);
        }
    }

    update = async (req: Request<{id: number}, any, User>, res: Response, next: NextFunction) => {
        const { id } = req.params;

        try {
            await this.service.update(id, req.body);
            return res.json({ message: 'Updated succesfully!' });
        } catch(err) {
            console.error('in update user:\n', err);
            next(err);
        }
    }

    remove = async (req: Request<{id: number}>, res: Response, next: NextFunction) => {
        const { id } = req.params;

        try {
            await this.service.remove(id);
            return res.json({ message: 'Deleted succesfully!', id });
        } catch(err) {
            console.error('in remove user:\n', err);
            next(err);
        }
    }
}

File: spring/src/main/java/hu/marko/szakdolgozat/spring/controller/UserController.java
package hu.marko.szakdolgozat.spring.controller;

import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import org.springframework.http.ResponseEntity;
import org.springframework.lang.Nullable;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import hu.marko.szakdolgozat.spring.controller.model.Message;
import hu.marko.szakdolgozat.spring.controller.model.User;
import hu.marko.szakdolgozat.spring.controller.model.wrapper.UserWrapper;
import hu.marko.szakdolgozat.spring.controller.model.wrapper.UsersWrapper;
import hu.marko.szakdolgozat.spring.service.UserService;
import lombok.AllArgsConstructor;

@AllArgsConstructor
@RestController
@RequestMapping("/api/users")
public class UserController {
  private UserService userService;

  @GetMapping("/page/{page}")
  public ResponseEntity<UsersWrapper> getAllUsers(@PathVariable("page") @NotNull Integer page,
      @NotNull @RequestParam("size") Integer size,
      @RequestParam("filt") @Nullable String filt, @RequestParam("ordr") @Nullable String ordr,
      @RequestParam("dir") @Nullable Boolean dir) {
    hu.marko.szakdolgozat.spring.service.model.PageModel<hu.marko.szakdolgozat.spring.service.model.User> sPageModel = userService
        .findByPageAndSizeAndFilterAndOrder(page, size, filt, ordr, dir);
    return ResponseEntity.ok().body(new UsersWrapper(
        StreamSupport.stream(sPageModel.getModels().spliterator(), false).map(User::new).collect(Collectors.toList()),
        sPageModel.getCount()));
  }

  @GetMapping("/{id}")
  public ResponseEntity<UserWrapper> getOneUser(@PathVariable("id") @NotNull Long id) {
    return ResponseEntity.ok().body(new UserWrapper(new User(userService.findOne(id))));
  }

  @PutMapping("/{id}")
  public ResponseEntity<Message> updateUser(@PathVariable("id") Long id, @RequestBody @Valid User user) {
    if (userService.update(id, user.toServiceUser())) {
      return ResponseEntity.ok().body(new Message("User is modified succesfully!"));
    }
    return ResponseEntity.badRequest().body(new Message("Couldn't modify user!"));
  }

  @DeleteMapping("/{id}")
  public ResponseEntity<Message> deleteUser(@PathVariable("id") Long id) {
    if (userService.remove(id) != null) {
      return ResponseEntity.ok().body(new Message("User is deactivated!"));
    }
    return ResponseEntity.badRequest().body(new Message("Couldn't deactivate user!"));
  }
}
